staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -1095,20 +976,19 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 			}
 		}
 
-		structural, err := structuralschema.NewStructural(schema)
-		if err == nil {
-			compResults, err := cel.Compile(structural, isRoot, cel.PerCallLimit)
+		if celContext.HasValidationRules() && celContext.TypeInfo() != nil {
```

Comment:
this is inside an `if len(schema.XValidations) > 0 {` block... why do we need `celContext.HasValidationRules()`?

Reply:
if `len(schema.XValidations) > 0`, and `celContext.TypeInfo() == nil`, shouldn't that be an error, rather than silently not compiling/honoring their validation rules? (this looks like a pre-existing issue since we ignored validations when `structuralschema.NewStructural(schema)` returned an error)

Reply:
Right.  I also found this when following through on some of the above comments about error handling.